DEV_GUIDE.md

```diff
@@ -0,0 +1,163 @@
+import org.scalaexercises.definitions._
+
+/** This line should describe your library shortly, and will appear as its description in the main Scala Exercises website. (2)
+  *
+  * @param name library_name (3)
+  */
+object MyLibrary extends Library {
+  override def owner = "owner" (4)
+  override def repository = "github_repository" (5)
+
+  override def color = Some("#HEX_COLOR") (6)
+
+  (7)
+  override def sections = List(
+    SectionA,
+    SectionB,
+    ...
+    SectionN
+  )
+}
+```
+
+Let's dig into each one of these points:
+
+* (1) A package name for your exercises library. 
+* (2) Inside the first ScalaDoc block, we define the description of the library as it'll appear in the list of the available libraries.
+* (3) Also, this ScalaDoc block defines the title of the library. It'll also be used as the path suffix for its URL. If you need to use spaces, use an underscore symbol (i.e.: "STD_lib")
+* (4) Owner username or organization in GitHub.
+* (5) Name of the repository that will contain the library.
+* (6) Color (in hexadecimal code format) that will identify the library in the website.
+* (7) A list that contains every section you want to include in this library. Every item defined here is a Scala object containing the exercises per section.
+
+####Creating a new section
+
+As previously mentioned, exercises are organized in sections that will be based on a Scala `object`. A good practice would be to have each section in a separate Scala file. So basically, the basic flow to create new exercises would be:
+
+1. Define the section `object` (we'll go through this process in a moment).
+2. Include a reference to this object in the `sections` list in the library `object` described in the previous section.
+
+How does a section look like?
+
+```scala
+/** @param name section_title (1)
+*/
+object MySection extends FlatSpec with Matchers with org.scalaexercises.definitions.Section (2) {
+    // exercises definition (3)
+}
+```
+
+* (1) As when defining the library, you need to use ScalaDoc to define the name of the section. Use an underscore symbol to include spaces.
+* (2) Each section `object` should inherit from `org.scalaexercises.definitions.Section`. `FlatSpec` and `Matchers` are traits from the ScalaTest library, the framework Scala Exercises uses to check the results.
+* (3) Inside the section `object` we can start defining the exercises, as it'll be shown in the following section.
+
+
+####Creating new exercises
+
+The content for each exercise is defined in a ScalaDoc block, with some specific particularities.
+
+```scala
+/** = Title = (1)
+*
+* Text describing background about the exercise, can be as long as needed. (2)
+*
+* {{{
+*   // code block (3)
+* }}}
+*
+* Documentation can be broken in as many paragraphs as needed.
+*/
+def functionAssert(res0: Boolean): Unit {
+    true shouldBe res0  // (4)
+}
+```
+
+* (1) You can define titles in the documentation for your library wrapping it inside a pair of equal signs (=).
```

Comment:
This is scaladoc specific and not sure we need to mention it here as we don't look for equal signs specifically